search-books/src/main/java/nix/education/service/suggestions/solr/SolrSuggestionService.java

```diff
@@ -0,0 +1,11 @@
+package nix.education.service.suggestions.solr;
+
+import java.util.Map;
+import lombok.SneakyThrows;
+import org.apache.solr.client.solrj.SolrClient;
+
+public interface SolrSuggestionService {
+
+    @SneakyThrows
+    Map<String, Object> prepareAuthorSuggestionsResponse(String q, SolrClient solrClient);
```

Comment:
Try to make interface methods params as flexible as possible. I don't think we need to pass solrClient to the method. You can inject this bean to a specific implementation

Reply:
I changed the second argument to Object and after that I can oberride this method for Solr and Elasticsearch.

Reply:
Changed the second argument of this method to Object.